api/src/main/java/org/pmiops/workbench/reporting/ReportingServiceImpl.java

```diff
@@ -36,8 +35,14 @@ public ReportingServiceImpl(
     this.reportingVerificationService = reportingVerificationService;
   }
 
+  private <T extends ReportingBase> void uploadBatchesForTable(
```

Comment:
Extracting this into a method allows us to remove the casting and SuppressWarnings.  My best guess as to why this works is that `T` is fully defined here (e.g. ReportingUser) instead of "any subclass of ReportingUser".s.